### PR TITLE
Make Lanth Line Better (mainly Samarium)

### PR DESCRIPTION
**Pull request by Nxer**
### New Samarium Part ###

- Oxidizing roasting , acid washing and Precipitate with oxalic acid , HCL wash to get Impure Samarium(III) Chloride Dust in the end .
- In IV - LuV tier , fix SmCl3 with NaCl then Electrolyze the blend , get Samarium dust , and recover others .
- In ZPM tier , use Lanthanum dust reduce SmCl3 in the Distillation Tower to get molten Samarium , La has been in Impure LaCl3 dust when output . Centrifuge it to get 'Rarer Earth Residue'(Something about the rarest Lanthanides , uncomplete now) and LaCl3 dust for recycling. 
- In UV tier , after Got the Nanites , use the Nanites' POWER to flexibly get any Lanthanides that has been got . The Extracting Nano Resin's recipes maybe need to rebalance.
- After having the Nanites' method , there is a method to increase production of Crystal SoC that use a new material 'Cerium-doped Lutetium Aluminium Garnet (Ce:LuAG)' . 
- The Cerium part has been basically designed , but not written yet .

### CODE REVIEW

**Comment by Dream-Master** [comment]
@Nxer you need to run spotless

**Comment by chochem** [comment]
For such a big overhaul the PR description needs a lot more detail. including but not limited to:
1) What's your overall goal? calling it 'better' is meaningless.
2) Did you speak with the author of the line?
3) do you have a spreadsheet or flowchart?

**Comment by Nxer** [comment]
> For such a big overhaul the PR description needs a lot more detail. including but not limited to:
> 
> 1. What's your overall goal? calling it 'better' is meaningless.
Give every ores themselves' process choice 
> 2. Did you speak with the author of the line?
Just now
> 3. do you have a spreadsheet or flowchart?
Coming soon

**Comment by JodieRuth** [comment]
The recipe for lanthanide element lines nowadays is almost just a search of relevant web pages and writing casually, and only half of it is written. This modification at least makes it look a lot more reasonable (when the product has no added use)

**Comment by Nxer** [comment]
Lanthanides Samarium Part

**Comment by Nxer** [comment]
- Give Samarium Ore more useable process choice .
- Only new recipes , no one old recipes has been changed or removed .
- Also makes some Advanced recipes by tiers , if someone need here are more choice can choose .
- When UV tier after got Carbon Nanites , Ion-Extracting about Lanthanides Process Method has been unlocked . It's a IMBA method to conveniently get any Lanthanides from any Lanthanides ores , just need that the Lanthanides has been got .
- Make a new way when got lot of Lutetium to craft Crystal SoC fastly . The number here need rebalance .
- Maybe the title should be "more finished" , it was the words "更完善" translated by machine .

File: src/main/java/com/elisis/gtnhlanth/loader/RecipeLoader.java [hunk 1]
```diff
@@ -978,6 +981,931 @@ public static void loadLanthanideRecipes() {
...
+        // 2B DilutedSamariumRareEarthSolution + 3B Oxalate =EV@10s= 5 ImpureSamariumOxalate + 0.1*2 LepersonniteDust +
+        // 50L
+        // MuddySamariumRareEarthSolution
+        GT_Values.RA.stdBuilder().itemInputs(GT_Utility.getIntegratedCircuit(13))
+                .fluidInputs(
+                        DilutedSamariumRareEarthSolution.getFluidOrGas(2000),
+                        MyMaterial.oxalate.getFluidOrGas(3000))
+                .itemOutputs(
+                        ImpureSamariumOxalate.get(OrePrefixes.dust, 5),
+                        GT_ModHandler.getModItem(Mods.GTPlusPlus.ID, "itemDustLepersonnite", 1))
```
Comment (Elisis) [inline src/main/java/com/elisis/gtnhlanth/loader/RecipeLoader.java]: Could you make this Samarskite instead?

**Review by Elisis — COMMENTED** [review]
If you change the IDs you're going to need to add the range to the list here: https://github.com/GTNewHorizons/bartworks/blob/56475bd9d13816fc70325d25349ec0cbaa1c097f/src/main/java/com/github/bartimaeusnek/bartworks/system/material/WerkstoffLoader.java#L183

File: src/main/java/com/elisis/gtnhlanth/loader/RecipeLoader.java (hunk 1 above)
Comment (Nxer) [reply src/main/java/com/elisis/gtnhlanth/loader/RecipeLoader.java]: Replace Lepersonnite with DephosphatedSamariumConcentrate or GT Rare Earth ? or something about other line example MonaziteRareEarthSediment , or delete it .

**Review by Elisis — APPROVED** [review]

**Comment by Elisis** [comment]
Still not entirely sure about the balance but it seems fairly reasonable as is

**Comment by Impos913** [comment]
> Only new recipes , no one old recipes has been changed or removed .

Screenshot_2023-09-29-09-38-58-154_com android chrome-edit
 Why nerf europium output again?

**Comment by Nxer** [comment]
> Why nerf europium output again?

5*Eu2O3 + 1*Eu =  4*EuO
                               ↓
5*Eu2O3 + 1*Eu =  6*EuO

Actually it's buff.... 

(I'm being too absolute)

**Pull request merged**